### ISSUE TITLE
[Bug] No camera is accessible in any page

### ISSUE THREAD

**Issue by dbenjaminmiller** [mozilla-mobile/fenix#18368]
## Steps to reproduce

1. Install Firefox.
2. Go to a web page that uses the webcam (like https://www.cam-recorder.com/).
3. I now get an Android system prompt for camera and microphone if I have not already granted permission.
4. I don't get a Firefox-level prompt.
5. No permission is ever given to the site to access any camera and I see no way to allow myself to grant that access.

### Expected behavior

Firefox should detect my phone's camera and prompt for its use when appropriate.

### Actual behavior

Firefox does not recognize my phone's camera.

### Device information

* Sony Xperia 1ii, Android 11
* Firefox for Android version: 86, but tested on 84 as well as current nightly and experienced the same problem

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-9973)

**Comment by Mikaka27** [comment]
Cannot reproduce on Galaxy j7 2017 (android 9). Maybe something specific to android 11?

**Comment by dbenjaminmiller** [comment]
It also could definitely be specific to my hardware (I think this is more likely than an Android 11 issue). I can try testing on a different phone later as well. I will note that the back camera (as accessed through the FF QR reader) works, and the front camera works fine in other apps (including Bromite browser).

**Comment by sflorean** [comment]
I was able to reproduce this on RC 85 build, with Huawei P40 Lite (Android 11) but not with Pixel 3 (Android 11).
However on latest Nightly build, I was able to reproduce only once, and after starting with a fresh profile, the issue is not reproducible anymore.

**Comment by kbrosnan** [comment]
I expect this is fallout from 0e8dc61 / #17809

**Comment by dbenjaminmiller** [comment]
Still an issue for me. I can try to get debug logs for it. I think it may have to do with the level of API exposed in the camera. FF should fall back to whatever level of the standard camera API is available, or at worst try another camera if there is another listed. But let me see what the actual error is.

**Comment by Photogad** [comment]
I am also having this issue, for many months now. Camera was working fine last year, but got broke at some point with either an Android OS update or a Fenix Update. Can't do any video calls, telemed appointments, nothing.

Chrome works fine. Fenix does not.  

![Screenshot_20210917-121403_Iceraven](https://user-images.githubusercontent.com/18505956/133821344-dcf2db10-e7f7-43b9-b511-4158d3034f32.jpg)
![Screenshot_20210917-121518_Chrome](https://user-images.githubusercontent.com/18505956/133821346-b321b9bc-7cbc-406e-9ed4-08b81c65d851.jpg)

Chrome always asks for camera permission. Fenix sometimes asks for camera permission (and doesn't work when I grant it), but most of the time it doesn't ask for permission at all.  Yes, in Android system app settings, permission is granted. Clear cache/data no help. Uninstall / reinstall app doesn't help.

I am using a Samsung Note10+ 5G, Android 11. My wife has same phone, and same problem on hers.  I also tried on a friend's Samsung S20+, same problem. Tried on another friend's Pixel (not sure model), on Android 11, and it worked on that.

Possibly a Samsung device issue? Not sure. But as it is now, cannot use or rely on Firefox as my default browser.

**Comment by dbenjaminmiller** [comment]
The issue definitely has to do with front/back camera, as well as the particular device. My device (the Xperia 1ii) fully exposes Camera3 for the back main camera and does not for the front-facing camera. (It provides only standard access.) I'm able to scan QR codes, so the back camera works, but not able to use my face.

If Camera3 access fails, FF needs to fall back.

**Comment by Photogad** [comment]
Yes, I can scan QR code.  It's only when a webpage is accessing the camera
that it doesn't work. The app itself can access camera.

**Comment by Photogad** [comment]
> The issue definitely has to do with front/back camera, as well as the particular device. My device (the Xperia 1ii) fully exposes Camera3 for the back main camera and does not for the front-facing camera. (It provides only standard access.) I'm able to scan QR codes, so the back camera works, but not able to use my face.
> 
> If Camera3 access fails, FF needs to fall back.

Yes, I can scan QR code. It's only pages themselves that can't access camera.  The app itself can access camera properly. I suppose webpage camera defaults to feont facing selfie camera.

**Comment by Photogad** [comment]
@sflorean @kbrosnan  any updates to this bug?

**Comment by dbenjaminmiller** [comment]
I do not use an Xperia anymore. I now use a Pixel, and there is no issue. It definitely has to do with some hardware and not others, perhaps relating to the Camera2 API?

**Comment by Nonius** [comment]
I've upgraded to Android 12, Firefox 96.3.1, still Samsung Galaxy note 10+, and it still doesn't work.

**Comment by amocirean** [comment]
I was able to reproduce the issue on the Huawei P40 Lite device, running Android 10. This issue occurs on Firefox Nightly, 18th of April's latest to be more specific.
In my specific case, no camera prompt is requested. Even if the permissions are given to the app, the camera still does not work.
This was tested on [this](https://mozilla.github.io/webrtc-landing/gum_test.html) page.
App Version: Nightly 101.0a1.

Notes:  I've tested the [page that you provided](https://www.cam-recorder.com/)) and the prompt was showing up.
